File: examples/nodejs/src/generateVcMessage.js
import { post } from "./lib/sock.js";

// PLEASE WRITE destination_did, messageJson, AND operation_tag.
const destination_did =
  "did:nodex:test:DummyDummyDummyDummyDummyDummyDummyDummyDummyD";
const messageJson = {
  message: {
    string: "value",
    number: 1,
    boolean: true,
    array: ["foo", "bar", "baz"],
    map: { key: "value" },
  },
};
const operation_tag = "test-operation-tag";

(async () => {
  const message = JSON.stringify(messageJson, null, 4);
  const response = await post("/create-verifiable-message", {
    destination_did,
    message,
    operation_tag,
  });

  console.log('\nPlease paste below to "verifyVcMessage.js".\n');
  console.log(response);
})();

File: examples/python/src/generate_vc_message.py
import json
import pprint
from platform_os import is_windows


if is_windows():
    from request import post
else:
    from sock import post


# PLEASE WRITE destination_did, message, AND operation_tag.
destination_did = (
    "did:nodex:test:DummyDummyDummyDummyDummyDummyDummyDummyDummyD"
)
message = {
    "message": {
        "string": "value",
        "number": 1,
        "boolean": True,
        "array": ["foo", "bar", "baz"],
        "map": {"key": "value"},
    }
}
operation_tag = "test-operation-tag"


payload = {
    "destination_did": destination_did,
    "message": json.dumps(message),
    "operation_tag": operation_tag,
}

json_response = post("/create-verifiable-message", payload)

print("The response is as follows.\n")
print(json_response)

print('\nPlease paste below to "verify_vc_message.py".\n')
pprint.pprint(json.loads(json_response), sort_dicts=False)
